### PR TITLE
Fixes #193

### LINKED ISSUE #193: complicated IWDRs & symlinks
Adapt https://github.com/common-workflow-language/cwltool/issues/1029 into a conformance test

### PR DESCRIPTION
**Pull request by GlassOfWhiskey**

### CODE REVIEW

**Comment by GlassOfWhiskey** [comment]
Here automated preview fails because
```
fatal: destination path '/opt/build/repo/temp//cwl-website' already exists and is not an empty directory.
```
It seems not related to the PR, but with something weird in the CI file system

File: tests/iwd/test-index.yaml [hunk 1]
```diff
@@ -297,3 +297,23 @@
       "checksum": "sha1$57f77b36009332d236b52b4beca77301b503b27c"
       "size": 268866
   tags: [ initial_work_dir, command_line_tool ]
+
+- id: iwd-symlinks
+  tool: iwd-symlinks.cwl
+  doc: "Test the use of symlinks in the InitialWorkDirRequirement"
```
Comment (mr-c) [inline tests/iwd/test-index.yaml:303]: Where is the symlink?

Comment (GlassOfWhiskey) [reply tests/iwd/test-index.yaml:303]: In the implementation, and in the title of the issue XD

Comment (mr-c) [reply tests/iwd/test-index.yaml:303]: Cwltool might be using a symlink, but that doesn't mean that other implementations would.

https://github.com/common-workflow-language/cwltool/issues/1029 outlines a more robust test

Comment (GlassOfWhiskey) [reply tests/iwd/test-index.yaml:303]: That test is exactly what I implemented here, just removing unnecessary parts and splitting CLTs.
If you want we can rename symlinks to another thing, but symlinks are not present in the original test, neither.

Comment (mr-c) [reply tests/iwd/test-index.yaml:303]: I see, thanks. Yes, the description needs updating. Can you confirm that this conformance test triggers the same bug in the old version of cwltool?

**Comment by GlassOfWhiskey** [comment]
Anything to do here?

File: tests/iwd/test-index.yaml (hunk 1 above)
Comment (mr-c) [reply tests/iwd/test-index.yaml:304]: Please remove the word `symlink` from this PR, unless you want to mention that cwltool had a bug related to complicated IWDRs & its internal use of symlinks

**Review by mr-c — CHANGES_REQUESTED** [review]

**Review by mr-c — APPROVED** [review]

**Pull request merged**